ts/features/mvl/screens/__test__/MvlRouterScreen.test.tsx

```diff
@@ -0,0 +1,168 @@
+      });
+    });
+
+    describe("And the mvlDetailsLoad is successful ", () => {
+      it("Should render the MvlDetailsScreen", () => {
+        const globalState = appReducer(
+          undefined,
+          applicationChangeState("active")
+        );
+        const store = createStore(appReducer, globalState as any);
+        const render = renderComponent(store);
+
+        render.store.dispatch(mvlDetailsLoad.success(mvlMockData));
+
+        expect(
+          render.component.queryByTestId("MvlDetailsScreen")
+        ).not.toBeNull();
+      });
+    });
+  });
+  describe("When the screen is rendered with a failure in feature.mvl.byId", () => {
+    it("Should render the MvlLoadingScreen", () => {
+      const globalState = appReducer(
+        undefined,
+        applicationChangeState("active")
+      );
+      const store = createStore(appReducer, globalState as any);
+      store.dispatch(
+        mvlDetailsLoad.failure({
+          kind: "generic",
+          id: mvlMockId,
+          value: new Error("An error")
+        })
+      );
+      const render = renderComponent(store);
+
+      expect(render.component.queryByTestId("MvlLoadingScreen")).not.toBeNull();
+    });
+
+    it("Should dispatch a loading request", () => {
+      const globalState = appReducer(
+        undefined,
+        applicationChangeState("active")
+      );
+      const store = createStore(appReducer, globalState as any);
+      store.dispatch(
+        mvlDetailsLoad.failure({
+          kind: "generic",
+          id: mvlMockId,
+          value: new Error("An error")
+        })
+      );
+      const mockStore = configureMockStore<GlobalState>();
+      const finalStore: ReturnType<typeof mockStore> = mockStore({
+        ...store.getState()
+      } as GlobalState);
+      renderComponent(finalStore);
+
+      expect(finalStore.getActions()).toStrictEqual([
+        mvlDetailsLoad.request(mvlMockId)
+      ]);
+    });
+  });
+
+  describe("When the screen is rendered with a success in feature.mvl.byId", () => {
+    it("Should render the MvlDetailsScreen", () => {
+      const globalState = appReducer(
+        undefined,
+        applicationChangeState("active")
+      );
+      const store = createStore(appReducer, globalState as any);
+      store.dispatch(mvlDetailsLoad.success(mvlMockData));
+      const render = renderComponent(store);
+
+      expect(render.component.queryByTestId("MvlDetailsScreen")).not.toBeNull();
+    });
+
+    it("Should NOT dispatch a loading request", () => {
+      const globalState = appReducer(
+        undefined,
+        applicationChangeState("active")
+      );
+      const store = createStore(appReducer, globalState as any);
+      store.dispatch(mvlDetailsLoad.success(mvlMockData));
+      const mockStore = configureMockStore<GlobalState>();
+      const finalStore: ReturnType<typeof mockStore> = mockStore({
+        ...store.getState()
+      } as GlobalState);
+      renderComponent(finalStore);
```

Comment:
I see you'd rather reach the desired state via actions rather than passing it from the beginning: keep in mind that at this point it becomes an **integration** test since you are also testing the reducers. It's fine as long as you know it and it is what you want.

Nonetheless, I'm more in favor of treating these as unit test: pass a well-defined and hardcoded state and test the render function only.

Reply:
Thanks for pointing this out. In this case was intentational as I prefer, whenever I can, to have integration tests because I think they bring more value in order to "simulate" multiple pieces interacting.